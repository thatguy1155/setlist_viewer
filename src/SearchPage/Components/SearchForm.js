import React, { useState } from 'react';
import { Link, useHistory } from 'react-router-dom';

export default function SearchForm(props) {
  const [artist, setArtist] = useState('');
  const [song, setSong] = useState('');
  const { search } = props;
  const history = useHistory();

  const readyForApi = (string) => string.toLowerCase().replace(' ', '%20');

  const handleChangeArtist = (event) => {
    setArtist(event.target.value);
  };
  const handleChangeSong = (event) => {
    setSong(event.target.value);
  };

  const handleClick = () => {
    const apiReadyArtist = readyForApi(artist);
    const apiReadySong = readyForApi(song);
    search({ artist: apiReadyArtist, song: apiReadySong });
    history.push('/results');
  };

  return (
    <div className="search-form">
      <label htmlFor="artist">
        Artist:
        <input type="text" id="artist" data-testid="artist" className="search-bar" onChange={handleChangeArtist} />
      </label>
      <label htmlFor="song">
        Song:
        <input type="text" id="song" data-testid="song" className="search-bar" onChange={handleChangeSong} />
      </label>
      <button type="button" className="submit-button" onClick={handleClick}>Submit</button>
    </div>
  );
}
