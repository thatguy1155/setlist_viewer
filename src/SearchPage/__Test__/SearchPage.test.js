import {
  act, render, screen, fireEvent,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import SearchForm from '../Components/SearchForm';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

describe('My SearchPage', () => {
  const callback = jest.fn();
  const history = createMemoryHistory();
  history.push('/');
  const component = renderer.create(
    <Router>
      <SearchForm artistName="" search={callback} />
    </Router>,
  );

  it('should render the Search Page', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  // it('asks to seach for artist', () => {
  //   // await screen.findByText(/Signed in as/);
  //   render(<SearchPage artistName="" search={callback} />);
  //   expect(
  //     screen.getByText(/Search For An Artist/),
  //   ).toBeInTheDocument();
  // });
  // it('asks to seach for song', () => {
  //   // await screen.findByText(/Signed in as/);
  //   render(<SearchPage search={callback} />);
  //   expect(
  //     screen.getByText(/Search For An Song by Bob Dylan/),
  //   ).toBeInTheDocument();
  //   // unmount();
  // });
  it('should return string ready for the API', () => {
    const form = render(<SearchForm value="" search={callback} />);
    fireEvent.change(form.getByTestId('artist'), {
      target: { value: 'Grateful Dead' },
    });
    fireEvent.change(form.getByTestId('song'), {
      target: { value: 'truckin' },
    });
    act(() => {
      userEvent.click(screen.getByRole('button'));
    });

    expect(callback).toHaveBeenCalledTimes(1);
    expect(callback).toHaveBeenCalledWith({ artist: 'grateful%20dead', song: 'truckin' });
  });
});
