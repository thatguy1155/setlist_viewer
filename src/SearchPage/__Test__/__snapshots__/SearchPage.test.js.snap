// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My SearchPage should render the Search Page 1`] = `
<div
  className="search-form"
>
  <label
    htmlFor="artist"
  >
    Artist:
    <input
      className="search-bar"
      data-testid="artist"
      id="artist"
      onChange={[Function]}
      type="text"
    />
  </label>
  <label
    htmlFor="song"
  >
    Song:
    <input
      className="search-bar"
      data-testid="song"
      id="song"
      onChange={[Function]}
      type="text"
    />
  </label>
  <button
    className="submit-button"
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;
